src/main/java/socketIO/ClientSocketInput.java

```diff
@@ -5,32 +5,37 @@
 import java.io.InputStreamReader;
 import java.io.InputStream;
 import java.net.Socket;
+import java.util.Scanner;
 
 public class ClientSocketInput implements IClientSocketInput {
 
   private Socket clientSocket;
+  private InputStream inputStream;
+  private Scanner scanner;
 
-  public ClientSocketInput(Socket clientSocket) {
+  public ClientSocketInput(Socket clientSocket) throws IOException {
     this.clientSocket = clientSocket;
+    this.inputStream = clientSocket.getInputStream();
+    this.scanner = new Scanner(inputStream, "UTF8");
   }
 
   public String getRawRequestString() throws IOException {
     StringBuilder request = new StringBuilder();
-    InputStreamReader isr = new InputStreamReader(clientSocket.getInputStream());
-    BufferedReader reader = new BufferedReader(isr);
-    String line = reader.readLine();
+    String line = scanner.nextLine();
     while (!line.isEmpty()) {
       request.append(line);
       request.append("\n");
-      line = reader.readLine();
+      line = scanner.nextLine();
     }
     return request.toString();
   }
 
   public byte[] getBytes(int length) throws IOException {
-    InputStream stream = clientSocket.getInputStream();
-    byte[] data = new byte[length];
-    stream.read(data);
-    return data;
+    scanner.useDelimiter("");
```

Comment:
I'm not entirely sure that scanner is the right class that we wnt to use here, especially since we're readying byte by byte.

Jake Nations is your man for this type of question :)